test/blackbox-tests/test-cases/unreadable-target.t/run.t

```diff
@@ -0,0 +1,22 @@
+  $ cat > dune-project <<EOF
+  > (lang dune 2.9)
+  > EOF
+
+  $ cat > dune <<EOF
+  > (rule
+  >   (targets a b)
+  >   (action (bash "echo content > a; chmod -r a; ln -s foo b")))
+  > 
+  > EOF
+
+  $ dune build b
+  File "dune", line 1, characters 0-84:
+  1 | (rule
+  2 |   (targets a b)
+  3 |   (action (bash "echo content > a; chmod -r a; ln -s foo b")))
+  Error: Error trying to read targets after a rule was run:
+  [ ("default/b",
+    "Unix.Unix_error(Unix.ENOENT, \"stat\", \"_build/default/b\")")
+  ; ("default/a", "Sys_error(\"_build/default/a: Permission denied\")")
```

Comment:
We generally try to avoid showing raw exceptions to users when it is a problem on their side. Here, we only catch `Sys_error` and `Unix_error` exceptions. Since `Sys_error` uses the formatting `<fname>: <unix error message>`, we could consistently use this formatting and print instead:


```
- _build/default/b: No such file or directory
- _build/default/a: Premission denied
```

BTW, I would expect `b` to be listed as a missing target

Reply:
`b` is a broken symlink. I definitely want an action producing a broken symlink to be reported differently from one producing nothing at all. It seems reasonable to write a few extra lines of code to detect that. Does that sound good?

Reply:
I missed that `b` was a broken symlink. Yes, I agree that they should be reported differently. A few extra lines of code to detect that seems good to me.

Reply:
I pushed the changes. The generic Unix_error code is no longer covered by tests because the cases to trigger it are getting ridiculous, but I tested it by removing the special case for broken symlink and it look good too:

```
- b: stat: No such file or directory
```